pkg/tfbridge/config_encoding.go

```diff
@@ -121,126 +113,90 @@ func (*ConfigEncoding) zeroValue(typ shim.ValueType) resource.PropertyValue {
 	}
 }
 
-func (enc *ConfigEncoding) unmarshalOpts() plugin.MarshalOptions {
-	return plugin.MarshalOptions{
+func (enc *ConfigEncoding) UnmarshalProperties(props *structpb.Struct) (resource.PropertyMap, error) {
+	m, err := plugin.UnmarshalProperties(props, plugin.MarshalOptions{
 		Label:        "config",
 		KeepUnknowns: true,
 		SkipNulls:    true,
 		RejectAssets: true,
-	}
-}
-
-// Like plugin.UnmarshalPropertyValue but overrides string parsing with convertStringToPropertyValue.
-func (enc *ConfigEncoding) unmarshalPropertyValue(key resource.PropertyKey,
-	v *structpb.Value) (*resource.PropertyValue, error) {
-
-	opts := enc.unmarshalOpts()
-
-	pv, err := plugin.UnmarshalPropertyValue(key, v, enc.unmarshalOpts())
+	})
 	if err != nil {
-		return nil, fmt.Errorf("error unmarshalling property %q: %w", key, err)
-	}
-	shimType, gotShimType := enc.fieldTypes[key]
-
-	// Only apply JSON-encoded recognition for known fields.
-	if !gotShimType {
-		return pv, nil
+		return nil, fmt.Errorf("failed to unmarshal to provider values: %w", err)
 	}
 
-	var jsonString string
-	var jsonStringDetected, jsonStringSecret bool
+	return enc.UnfoldProperties(m)
+}
 
-	if pv.IsString() {
-		jsonString = pv.StringValue()
-		jsonStringDetected = true
-	}
+func (enc *ConfigEncoding) UnfoldProperties(m resource.PropertyMap) (resource.PropertyMap, error) {
+	result := make(resource.PropertyMap, len(m))
+	for _, pk := range m.StableKeys() {
+		v := m[pk]
+
+		// If we don't have type information on the field, we don't attempt to do
```

Comment:
There's too many changes to config_encoding for this PR. There's semantic changes here it looks like, adding to the unease. I'd like to request making these changes step-by-step for easier review to understand why this is happening, so we can double-check this is being tested properly and makes sense.

There is this PR by @blampe 
https://github.com/pulumi/pulumi/pull/15032 that would still be good to land to not use this encoding at all. 

I'm trying to recall the rollut plan there. Once the bridge makes sure that fix is rolled out to the code generators, subsequent versions might not ever need to deal with this encoding at all and may drop the code cold-turkey.